src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -897,19 +766,32 @@ private boolean isDefaultExport(TypedVar symbol) {
 
   /** For platform externs we skip emitting, to avoid collisions with lib.d.ts. */
   private boolean isPlatformExtern(String filePath, String symbolName) {
-    // Some symbols are not yet provided by lib.d.ts, so we have to emit them.
-    if (PLATFORM_SYMBOLS_MISSING_IN_TYPESCRIPT.contains(symbolName)) return false;
-    // This matches our test setup.
-    if (filePath.startsWith("externs.zip//")) return true;
+    // This file is built in to Clutz for testing, so special-case it here.
     String fileName = new File(filePath).getName();
-    // This loosely matches the filenames from
-    // https://github.com/google/closure-compiler/tree/master/externs
-    return PLATFORM_EXTERNS_FILENAMES.contains(fileName)
-        || fileName.startsWith("es")
-        || fileName.startsWith("gecko_")
-        || fileName.startsWith("w3c_")
-        || fileName.startsWith("ie_")
-        || fileName.startsWith("webkit_");
+    if (fileName.equals("es6_min.js")) return true;
+
+    // If asked about a symbol like "window.Array", perform the below lookups on just
+    // "Array".
+    if (symbolName.startsWith("window.")) {
+      symbolName = symbolName.substring("window.".length());
+    }
+
+    // We're asked about e.g. "Array.from", which we should treat as a platform extern.
+    // The PlatformSymbols lists just contain the toplevel name like "Array", so strip after
+    // the first dot.
+    int dotPos = symbolName.indexOf('.');
+    if (dotPos > 0) {
+      symbolName = symbolName.substring(0, dotPos);
+    }
+
+    // Don't emit externs for Closure types that nobody uses.
```

Comment:
This line goes against the method name 'isPlatformExtern', can we change the method name, or at least put a big warning there, that for the purposes of optimization we kinda lie here.

If we make externs_list incremental too one day, we should revisit this.

Reply:
Renamed, not super happy with the name.